app/main.py

```diff
@@ -1,2 +1,55 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: str, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor_threshold = 0.75
+        self.table = [[] for _ in range(self.capacity)]
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor_threshold:
+            self._resize()
+
+        index = self._get_index(key)
+        for node in self.table[index]:
+            if node.key == key:
+                node.value = value
+                return
+
+        self.table[index].append(self.Node(key, value))
+        self.size += 1
+
+    def __getitem__(self, key: str) -> Any:
+        index = self._get_index(key)
+
+        for node in self.table[index]:
+            if node.key == key:
+                return node.value
+
+        raise KeyError(f"Key {key} not found.")
```

Comment:
Make sure to provide a message when an exception is raised. The KeyError here already includes a message, so this is correct.